### DISCUSSION TITLE
How to protect a page using Auth.js with Firebase (Firestore DB) adapter?

### DISCUSSION THREAD

**Discussion by Runkang10** [nextauthjs/next-auth#11304] [Help] [answered]
Hi everyone,
I'm trying to protect a page using Auth.js with Firebase (Firestore DB) adapter but i got problems with Database: `Edge runtime error` or unsupported modules like `os` or `fs` in Next.js + TypeScript.

Can anyone help me please 😢

**Comment by k-taro56** [comment]
Do you want it to work with Edge Runtime?
If not, check for `const runtime = 'edge'`.

**Comment by Runkang10** [comment, marked as answer]
Nevermind I forgot what was the issue I got but probably using Next.js Middleware by following the documentation at that time or calling database calls on client side...
